spyder/api/plugins.py

```diff
@@ -454,6 +744,7 @@ def refresh_plugin(self):
         """
         pass
 
+    # TODO: This method should be deprecated or rethinked
```

Comment:
This method is very important because it allows to share actions between the plugin and toolbars or main menus. So I don't agree with removing it.

Reply:
I know that it is being used and I understand how it is used, that is why I say rethink cause it is inconsistently used among different plugins and widgets.